presto-parquet/src/test/java/com/facebook/presto/parquet/writer/TestParquetWriter.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.parquet.writer;
+
+import com.facebook.presto.common.PageBuilder;
+import com.facebook.presto.common.type.Type;
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.common.type.IntegerType.INTEGER;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.io.Files.createTempDir;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static java.util.UUID.randomUUID;
+import static org.testng.Assert.fail;
+
+public class TestParquetWriter
+{
+    private File temporaryDirectory;
+    private File parquetFile;
+
+    @BeforeMethod
+    public void setup()
+    {
+        temporaryDirectory = createTempDir();
+        parquetFile = new File(temporaryDirectory, randomUUID().toString());
+    }
+
+    @AfterMethod
+    public void tearDown()
+            throws IOException
+    {
+        deleteRecursively(temporaryDirectory.toPath(), ALLOW_INSECURE);
+    }
+
+    @Test
+    public void testRowGroupFlushInterleavedColumnWriterFallbacks()
+    {
+        List<Type> types = ImmutableList.of(BIGINT, INTEGER, VARCHAR, BOOLEAN);
+        List<String> names = ImmutableList.of("col_1", "col_2", "col_3", "col_4");
+        ParquetWriterOptions parquetWriterOptions = ParquetWriterOptions.builder()
+                .setMaxPageSize(DataSize.succinctBytes(1000))
+                .setMaxBlockSize(DataSize.succinctBytes(15000))
+                .setMaxDictionaryPageSize(DataSize.succinctBytes(1000))
+                .build();
+        try (ParquetWriter parquetWriter = createParquetWriter(parquetFile, types, names, parquetWriterOptions, CompressionCodecName.UNCOMPRESSED)) {
```

Comment:
Better to test more compression methods since this change is about page size settings.

Reply:
The main purpose to have a chance to set 'maxDictionaryPageSize' is to reappear the bug when writing to parquet more faster. Or we could only use default 'maxDictionaryPageSize' as 1M, which means we should set bigger 'maxPageSize' and 'maxBlockSize' than this unit test case and write much more data as in real production environment to reappear the problem.